### PR TITLE
Implemented login/logout features

### LINKED ISSUE #6: Add login in functionality to Vuex store
## Acceptance criteria
* The auth store contains a login action, that commits the user object to the auth state.
* It should contain a mutation for signing in and out
* It should contain a state where a "session" is stored

### LINKED ISSUE #11: Hide admin button in header if signed in as a student
## Solution proposal
* Check the user privileges stored in the user object in the Vuex Store

```ts
state: {
  auth : userObject
}
```

```ts
userObject: {
 id: number,
 name: string,
 type: string,
 email: string,
 password: string
}
```

### LINKED ISSUE #12: Add logout functionality
## Acceptance criteria
* The header should contain a logout button when signed in
* Pressing the button should result in the vuex user object being cleared thus signing the user out
* The user should then be redirected to the home page

### PR DESCRIPTION
**Pull request by te4-adrian-almetun-smeds**
## What issue did you implement or fix?
Fixed #6
Fixed #11 
Fixed #12

## Summary
- Added auth module to store
- Added login functionality
- Added logout functionality
- Implemented functionality in to the header 
- Implemented basic functionality to the login component**

 ** Input the user ID into the email input to login otherwhise it defaults to user 1 
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by TE4-David-Sundqvist — APPROVED** [review]
👍 very good

File: lib/src/store/modules/users.ts [hunk 1]
```diff
@@ -81,6 +81,9 @@ export default {
     users: (state: any) => {
       return state.users;
     },
+    user: (state: any) => (userID: number) => {
```
[the remark this answers is not in the document]
Comment (te4-adrian-almetun-smeds) [reply lib/src/store/modules/users.ts:56]: I can't see a way not to duplicate this without it requiring a <b> lot </b> of effort, that isn't provided on the given timeline.

**Comment by te4-adrian-almetun-smeds** [comment]
WIll wait on @te4-tintin-wihlborg or @te4-henrik-stahl to also accept before merging.

**Review by te4-henrik-stahl — APPROVED** [review]
wow yes wow

**Pull request merged**